Admin/Adapters/FileSystem/FileSystemDal.ts

```diff
@@ -0,0 +1,60 @@
+import fs = require('fs');
+import Port = require('../../Ports/IDal');
+import path = require('path');
+
+class FileSystemDal implements Port.IDal {
+  /**
+   * Find all files inside a dir, recursively.
+   * @function getAllFilesInDirectoryRecursive
+   * @param  {string} dir Dir path string.
+   * @return {string[]} Array with all file names that are inside the directory.
+   */
+  public getAllFilesInDirectoryRecursive = (dir: string): string[] =>
```

Comment:
worth adding testing at this part?

Reply:
Agree, Will do :)

Reply:
should the test create files and delete them?

Reply:
You should mock fs I think